Web Inspector: Add Versioned Protocol for iOS 26.0 and macOS 26.0

AnujPanta1 opened the pull request:
#### fbe70bc762df3d69400b0c090659a5404b0250e5
<pre>
Web Inspector: Add Versioned Protocol for iOS 26.0 and macOS 26.0
<a href="https://bugs.webkit.org/show_bug.cgi?id=294546">https://bugs.webkit.org/show_bug.cgi?id=294546</a>
<a href="https://rdar.apple.com/143587423">rdar://143587423</a>

Reviewed by Devin Rousso.

* Source/WebInspectorUI/UserInterface/Controllers/AnimationManager.js:
(WI.AnimationManager.prototype.effectChanged):
* Source/WebInspectorUI/UserInterface/Controllers/TimelineManager.js:
(WI.TimelineManager.prototype.initializeTarget):
(WI.TimelineManager.prototype.disable):
(WI.TimelineManager.prototype._updateAutoCaptureInstruments):
* Source/WebInspectorUI/UserInterface/Models/Animation.js:
(WI.Animation.fromPayload):
(WI.Animation.prototype.effectChanged):
* Source/WebInspectorUI/UserInterface/Models/Instrument.js:
(WI.Instrument.startLegacyTimelineAgent):
(WI.Instrument.stopLegacyTimelineAgent):
* Source/WebInspectorUI/UserInterface/Models/ScriptInstrument.js:
(WI.ScriptInstrument.prototype.startInstrumentation):
(WI.ScriptInstrument.prototype.stopInstrumentation):
(WI.ScriptInstrument):
* Source/WebInspectorUI/UserInterface/Models/SourceMapResource.js:
(WI.SourceMapResource.prototype.get supportsScriptBlackboxing):
* Source/WebInspectorUI/UserInterface/Protocol/AnimationObserver.js:
(WI.AnimationObserver.prototype.effectChanged):
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/26.0/InspectorBackendCommands.js: Added.
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/26.0/InspectorBackendCommands.js: Added.
* Source/WebInspectorUI/UserInterface/Views/DOMTreeContentView.js:
(WI.DOMTreeContentView.prototype._populateConfigureDOMTreeBadgesNavigationItemContextMenu):
* Source/WebInspectorUI/Versions/Inspector-iOS-26.0.json: Added.
* Source/WebInspectorUI/Versions/Inspector-macOS-26.0.json: Added.

Canonical link: <a href="https://commits.webkit.org/296913@main">https://commits.webkit.org/296913@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/686eff124175ef2aeb46e7e78c005f85f16aa0a2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/109841 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/29499 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/19929 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/115862 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/60078 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/30177 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38087 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/83478 "") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/112789 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24052 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/98919 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/63938 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/23432 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/17065 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/59657 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/93422 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/17108 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/118654 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/36880 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/27328 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/92475 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/37253 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95184 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92298 "Found 2 new API test failures: /WebKitGTK/TestWebProcessExtensions:/webkit/WebKitWebView/web-process-crashed, /WebKitGTK/TestWebKitPolicyClient:/webkit/WebKitPolicyClient/new-window-policy (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23540 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37275 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15019 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/32726 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/36775 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/42245 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/36435 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/39777 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/38144 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebInspectorUI/UserInterface/Controllers/AnimationManager.js:
```diff
@@ -126,7 +126,7 @@ WI.AnimationManager = class AnimationManager
         if (!animation)
             return;
 
-        // COMPATIBILITY (iOS 18.X, macOS 15.X): `Animation.effectChanged` removed the `effect` parameter in favor of `Animation.requestEffect`.
```
there's also a lot more changes for `COMPATIBILITY (iOS X.Y, macOS X.Y)` that need to be updated

dcrousso commented on Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/26.0/InspectorBackendCommands.js:
```diff
@@ -0,0 +1,571 @@
+
+// Console
+InspectorBackend.registerDomain("Console", ["itml", "javascript", "page", "service-worker", "worker"]);
+InspectorBackend.registerEnum("Console.ChannelSource", {XML: "xml", JavaScript: "javascript", Network: "network", ConsoleAPI: "console-api", Storage: "storage", Appcache: "appcache", Rendering: "rendering", CSS: "css", Accessibility: "accessibility", Security: "security", ContentBlocker: "content-blocker", Media: "media", MediaSource: "mediasource", WebRTC: "webrtc", ITPDebug: "itp-debug", Privat ...
```
oh hmm apparently `MessageSource::AppCache` is still in use.  i think we may need to add this back to all the old protocol `.js`/`.json` in that case (i.e. undo these changes from [289820@main](https://commits.webkit.org/289820@main))

rcaliman-apple replied:
AppCache has been marked for deprecation in Safari since 2018:
https://bugs.webkit.org/show_bug.cgi?id=181764

Both Chrome and Firefox have deprecated the feature:
https://web.dev/articles/appcache-removal

We intentionally rewrote the older protocol snapshots to remove AppCache to prevent issues where Web Inspector doesn't have the frontend required to inspect an older device that happens to still use the deprecated feature: https://github.com/WebKit/WebKit/pull/39648#discussion_r1933144397

rcaliman-apple replied:
@AnujPanta1 there are some references to `appcache` that should be manually removed, like we did for the [previous snapshot patch](https://github.com/WebKit/WebKit/pull/41071#discussion_r1966341561)

dcrousso replied:
so even though we removed support for inspecting AppCache in [289820@main](https://commits.webkit.org/289820@main) there are still uses of `MessageSource::AppCache` inside WebCore.  as such, we need to keep around `Console.ChannelSource.AppCache` in order for the frontend to support those console logs

if we are OK with removing it because AppCache is deprecated (which was the logic behind prematurely removing frontend support in [289820@main](https://commits.webkit.org/289820@main)) then we'd need to change those cases in WebCore to use `MessageSource::Other`

rcaliman-apple replied:
Good catch! We should replace `MessageSource::AppCache` with `MessageSource:: Other` in WebCore so that we can also remove the ["appcache"channel source type from `Console.json`](https://github.com/WebKit/WebKit/blob/d6dcfd454746cdbb48de0e9b41fc10f03b0843e8/Source/JavaScriptCore/inspector/protocol/Console.json#L16) and avoid having to do this manual rewrite again.

That can be a separate patch. Filed: https://bugs.webkit.org/show_bug.cgi?id=294907

rcaliman-apple replied:
PR to replace `MessageSource::AppCache`: https://github.com/WebKit/WebKit/pull/47122

AnujPanta1 replied:
I removed the references to `appcache` and got those other `COMPATIBILITY (iOS X.Y, macOS X.Y) ` I didn't know those were apart of the PR as well thank you !

dcrousso commented on Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/26.0/InspectorBackendCommands.js:
```diff
@@ -0,0 +1,570 @@
+
+// Console
+InspectorBackend.registerDomain("Console", ["itml", "javascript", "page", "service-worker", "worker"]);
+InspectorBackend.registerEnum("Console.ChannelLevel", {Off: "off", Basic: "basic", Verbose: "verbose"});
```
there should still be a `ChannelSource` here, just without any mention of `AppCache`

rcaliman-apple replied:
Yeah, the latest change was too broad: [diff](https://github.com/WebKit/WebKit/compare/b8610497e2eaa23c810b15688750dd308a321e3e..df5c1366be692dd75ddb39137870b1a3468aa1dc#diff-8e228f13a146e14a7ea22884f954ee8cf24644861c92ce53b126799a3b5589d3)

dcrousso reviewed (commented):
there's actually one more comment
```js
// COMPATIBILITY (macOS X.Y, iOS X.Y): Debugger.setShouldBlackboxURL.sourceRanges did not exist yet.
```
that needs to be changed to
```js
// COMPATIBILITY (macOS 15.0, iOS 18.0): Debugger.setShouldBlackboxURL.sourceRanges did not exist yet.
```
if you dont mind doing that as well

AnujPanta1 committed:
Web Inspector: Add Versioned Protocol for iOS 26.0 and macOS 26.0
https://bugs.webkit.org/show_bug.cgi?id=294546
rdar://143587423

Reviewed by Devin Rousso.

* Source/WebInspectorUI/UserInterface/Controllers/AnimationManager.js:
(WI.AnimationManager.prototype.effectChanged):
* Source/WebInspectorUI/UserInterface/Controllers/TimelineManager.js:
(WI.TimelineManager.prototype.initializeTarget):
(WI.TimelineManager.prototype.disable):
(WI.TimelineManager.prototype._updateAutoCaptureInstruments):
* Source/WebInspectorUI/UserInterface/Models/Animation.js:
(WI.Animation.fromPayload):
(WI.Animation.prototype.effectChanged):
* Source/WebInspectorUI/UserInterface/Models/Instrument.js:
(WI.Instrument.startLegacyTimelineAgent):
(WI.Instrument.stopLegacyTimelineAgent):
* Source/WebInspectorUI/UserInterface/Models/ScriptInstrument.js:
(WI.ScriptInstrument.prototype.startInstrumentation):
(WI.ScriptInstrument.prototype.stopInstrumentation):
(WI.ScriptInstrument):
* Source/WebInspectorUI/UserInterface/Models/SourceMapResource.js:
(WI.SourceMapResource.prototype.get supportsScriptBlackboxing):
* Source/WebInspectorUI/UserInterface/Protocol/AnimationObserver.js:
(WI.AnimationObserver.prototype.effectChanged):
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/26.0/InspectorBackendCommands.js: Added.
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/26.0/InspectorBackendCommands.js: Added.
* Source/WebInspectorUI/UserInterface/Views/DOMTreeContentView.js:
(WI.DOMTreeContentView.prototype._populateConfigureDOMTreeBadgesNavigationItemContextMenu):
* Source/WebInspectorUI/Versions/Inspector-iOS-26.0.json: Added.
* Source/WebInspectorUI/Versions/Inspector-macOS-26.0.json: Added.

Canonical link: https://commits.webkit.org/296913@main
